src/sources/orbit.ts

```diff
@@ -0,0 +1,231 @@
+import { getAddress } from 'viem'
+import { z } from 'zod'
+import { SourceContext } from './source.js'
+import { env } from '../env.js'
+import { assert } from '@l2beat/backend-tools'
+import { nanoid } from 'nanoid'
+
+export function buildOrbitSource({ logger, db }: SourceContext) {
+  logger = logger.for('OrbitSource')
+
+  return async () => {
+    logger.info(`Fetching tokens from Orbit...`)
+
+    const networks = await db.network
+      .findMany({
+        include: {
+          rpcs: true,
+        },
+        where: {
+          orbitId: {
+            not: null,
+          },
+        },
+      })
+      .then((result) =>
+        result.map((r) => {
+          const { orbitId } = r
+          assert(orbitId, 'Expected orbitId')
+          return {
+            ...r,
+            orbitId,
+          }
+        }),
+      )
+
+    const res = await fetch(env.ORBIT_LIST_URL)
+    const data = await res.json()
+    const parsed = OrbitResult.parse(data)
+
+    logger.info('Upserting bridge info')
+
+    const { id: bridgeId } = await db.bridge.upsert({
+      select: { id: true },
+      where: {
+        name: 'Orbit',
+      },
+      create: {
+        id: nanoid(),
+        name: 'Orbit',
+      },
+      update: {},
+    })
+
+    let count = 0
+    const totalCount = parsed.tokenList.flatMap((token) => [
+      null,
+      Object.values(token.minters),
+    ]).length
+
+    for (const token of parsed.tokenList) {
+      logger.debug('Processing token', { symbol: token.symbol })
+      const sourceNetwork = networks.find(
+        (chain) => chain.orbitId && chain.orbitId === token.chain,
+      )
+
+      if (!sourceNetwork) {
+        logger.debug('No source network found', {
+          symbol: token.symbol,
+          chain: token.chain,
+        })
+        continue
+      }
+
+      logger.debug('Upserting source token', { symbol: token.symbol })
+
+      const { id: sourceTokenId } = await db.token.upsert({
+        select: { id: true },
+        where: {
+          networkId_address: {
+            networkId: sourceNetwork.id,
+            address: getAddress(token.address),
+          },
+        },
+        create: {
+          id: nanoid(),
+          networkId: sourceNetwork.id,
+          address: getAddress(token.address),
+        },
+        update: {},
+      })
+
+      await db.tokenMeta.upsert({
+        where: {
+          tokenId_source: {
+            tokenId: sourceTokenId,
+            source: 'orbit',
```

Comment:
We should use uppercase here to align with other sources.

Reply:
We'll solve that later, as we have this issue in various different places.